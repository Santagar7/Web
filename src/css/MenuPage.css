.menu{
  width: 100%;
  height: 100%;
  position: absolute;
  top:0;
}
.wrapper{
  width: 100%;
  height: calc(100%-60px);
}
#actions {
  font-family: Arial, Helvetica, sans-serif;
  border-collapse: collapse;
  width: 80%;
  max-height: 60%;
  margin: 40px 10%;
  background: white;
  border: solid silver 2px;
  border-radius: 15px;
  overflow: scroll;
}
#actions td, #actions th {
  border: 1px solid #ddd;
  padding: 8px;
}
#actions tr:nth-child(even){
  background-color: #f2f2f2;
}
#actions tr:hover {
  background-color: #ddd;
}
#actions th {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: left;
  background-color: #04AA6D;
  color: white;
}
.cost{
    background: rgb(253, 144, 144);
}
.profit{
    background: rgb(130, 247, 130);
}
.money{
  width: 40%;
  height: 50px;
  background: white;
  margin: 80px auto;
  text-align: center;
  border-radius: 15px;
  min-width: fit-content;
  border: solid silver 2px;
}
.money h1{
  margin: 0;
  line-height: 50px;
}
.panel{
  position:absolute;
  left: 0;
  right: 0;
  margin: 0 auto;
  width:100%;
  height:100%;
}
.panel input{
  width: 50% !important;
  margin: 15px 25%;
  padding: 5px;
  box-sizing: border-box;
  transition: 0.2s ease all;
  border:none;
  border-bottom: 2px solid silver;
  background: whitesmoke;
  font-size: 18px;
}
.panel input:focus{
  outline: none;
  border-bottom: 2px solid blue;
}
.panel input:hover {
  border-bottom: 2px solid blue;
}
.panel form button{
  margin: 10px 25%;
}
