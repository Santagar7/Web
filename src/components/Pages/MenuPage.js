import Navbar from '../Elements/Navbar'
import { useParams } from 'react-router-dom'
import React, { useEffect, useState } from 'react'

import axios from 'axios'
import '../../css/MenuPage.css'

axios.defaults.withCredentials = true
/* eslint-disable react/prop-types */

function Action ({ item }) {
  const cl = item.amount > 0 ? 'profit' : 'cost'
  const date = new Date(item.time)
  return (
        <tr>
            <td>{item.made_by}</td>
            <td>{item.reason}</td>
            <td>{date.toDateString()}</td>
            <td className={cl}>${item.amount}</td>
        </tr>
  )
}

function MenuPage () {
  const { index } = useParams()
  const [money, setMoney] = useState(0)
  const [actions, setActions] = useState([])
  const [refresh, setRefresh] = useState(true)
  useEffect(() => {
    const url = 'http://localhost:7000/'
    axios.get(url + `menu/${index}`).then((resp) => {
      setMoney(resp.data.money)
      setActions(resp.data.arr)
    })
  }, [refresh])
  return (
        <div className='menu'>
            <Navbar index={index} val={refresh} refresh={setRefresh}/>
            <div className="wrapper">
                <div className="money">
                    <h1 id="money">{`Current balance : ${money}`}</h1>
                </div>
                <table id="actions">
                    <thead>
                        <tr>
                            <th>Made by</th>
                            <th>Purpose/Source</th>
                            <th>Time</th>
                            <th>Amount</th>
                        </tr>
                    </thead>
                    <tbody>
                        {actions.map(item =>
                          (<Action item={item} key={item.time}/>))}
                    </tbody>
                </table>
            </div>
        </div>
  )
}
export default MenuPage
