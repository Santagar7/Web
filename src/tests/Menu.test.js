import { BrowserRouter } from 'react-router-dom';
import { expect, test } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MenuPage from '../components/Pages/MenuPage';

test('should render correctly', () => {
    const tree = renderer.create(<BrowserRouter><MenuPage/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

test('check current balance', () => {
    render(
        <BrowserRouter>
            <MenuPage />
        </BrowserRouter>,
    );

    const linkElement = screen.getByText(/Current balance/i);
    expect(linkElement).toBeInTheDocument();
});

test('check actions panel', () => {
    render(
        <BrowserRouter>
            <MenuPage />
        </BrowserRouter>,
    );
    const submit = screen.getByText(/Actions/i);
    fireEvent.click(submit);
    const linkElement = screen.getByText(/Create new Action/i);
    expect(linkElement).toBeInTheDocument();
});

test('check members panel', () => {
    render(
        <BrowserRouter>
            <MenuPage />
        </BrowserRouter>,
    );
    const submit = screen.getByText(/Members/i);
    fireEvent.click(submit);
    const linkElement = screen.getByText(/All Members/i);
    expect(linkElement).toBeInTheDocument();
});