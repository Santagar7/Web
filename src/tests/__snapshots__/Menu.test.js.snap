// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="menu"
>
  <div
    className="extra"
  >
    <div
      className="topnav"
    >
      <a
        className="active"
        onClick={[Function]}
      >
        Log Out
      </a>
      <a
        data-testid="members"
        href="/"
        id="members"
        onClick={[Function]}
      >
        Members
      </a>
      null:
      <a
        data-testid="budgets"
        href="/budgets"
        onClick={[Function]}
      >
        Budgets
      </a>
      null:
      <a
        data-testid="action"
        href="/"
        id="action"
        onClick={[Function]}
      >
        Actions
      </a>
    </div>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="money"
    >
      <h1
        id="money"
      >
        Current balance : 0
      </h1>
    </div>
    <table
      id="actions"
    >
      <thead>
        <tr>
          <th>
            Made by
          </th>
          <th>
            Purpose/Source
          </th>
          <th>
            Time
          </th>
          <th>
            Amount
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
